src/lib/writer-json-sarif.cc

```diff
@@ -276,10 +276,15 @@ void SarifTreeEncoder::appendDef(const Defect &def)
         shellCheckMap_[ruleId] = sm[2];
     }
 
-    if (def.cwe)
+    if (def.cwe) {
         // update CWE map
         cweMap_[ruleId] = def.cwe;
 
+        // encode per-warning CWE property
+        const object cweProp = {{ "cwe", "CWE-" + std::to_string(def.cwe) }};
```

Comment:
You can't move a `const` object because the type of `std::move(cweProp)` is `const object&&` which is not a proper candidate for the move assignment operator.
```suggestion
        object cweProp = {{ "cwe", "CWE-" + std::to_string(def.cwe) }};
```